### PR TITLE
Move charm library helper functions to their own modules.

### PR DESCRIPTION
**Pull request by lengau**
Fixes CRAFT-1557

### CODE REVIEW

File: tests/commands/test_store_charmlibs.py [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+        [
+            pytest.param("LIBID = 99", id="Non-string"),
+            pytest.param("LIBID = 'moño'", id="Non-ASCII"),
+            pytest.param("LIBID = ''", id="empty"),
+        ],
+    )
+    def test_libid_invalid(self, metadata_id):
+        """The ID is not really a string."""
+        test_path = self.create_lib(metadata_id=metadata_id)
+        match = r"Library .+ metadata field LIBID must be a non-empty ASCII string\."
```
Comment (facundobatista) [inline tests/commands/test_store_charmlibs.py:329]: Here and in other tests, do not build generic strings when checking for errors or other messages exposed to the user.

For example in this case, if we introduce a bug in the code and the error message is `Library foobarBROKEN metadata field LIBID must be a non-empty ASCII string.` this still will pass, and it shouldn't.

Yes, the chances are low, but we have to be extra careful with messages shown to the final user.

File: tests/commands/test_store_charmlibs.py [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+    )
+    def test_api_patch_invalid(self, lib_fields, match):
+        """Invalid values for API or patch versions."""
+        test_path = self.create_lib(**lib_fields)
+        with pytest.raises(CraftError, match=match):
+            get_lib_info(lib_path=test_path)
+
+    def test_metadata_api_different_path_api(
+        self,
+    ):
```
Comment (facundobatista) [inline tests/commands/test_store_charmlibs.py:311]: This should be one line.

File: tests/commands/test_store_charmlibs.py [hunk 3]
```diff
@@ -0,0 +1,331 @@
...
+            pytest.param(b"", ValueError, "invalid literal", id="empty-bytes"),
+        ],
+    )
+    def test_exceptions(self, raw_value, error_class, error_regex):
+        """Test various exceptions that can be passed through."""
+        with pytest.raises(error_class, match=error_regex):
+            get_positive_int(raw_value)
+
+
+class TestGetLibInfo:
```
Comment (facundobatista) [inline tests/commands/test_store_charmlibs.py:56]: We're trying to move away from class-structured tests in pytest. Is there specific functionality that you need here?

**Review by facundobatista — CHANGES_REQUESTED** [review]
Thanks for this work!

There's a problem with this PR, though. PRs should be small, so we can review them... "PRs that move stuff around" are intrinsically big, but it's fine because you don't have to review them

This PR is a) huge  b) has a lot of changes. While most of the changes I agree, there are some changes that need to be reverted (see inline comments).

Please, propose a PR with *just moving code around* (big, but no real review needed), and then after that, if you want to do changes, more PRs (small and reviewable).

Thanks!!

**Comment by facundobatista** [comment]
There quite some changes that are welcomed there, and I would expect in another PR (e.g. using parametrization to merge tests that are too similar).

That said, of course *some* changes are ok in the "move stuff around", like different imports, or removing underscores because of transforming something private in public (so it can live in other module).

Thanks!

**Comment by lengau** [comment]
@facundobatista Agreed. I'm going to close this PR and redo it as two.

**Pull request closed without merging**